modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2236,7 +2206,67 @@ def sort_function(df):  # pragma: no cover
             new_partitions,
             index,
             shuffling_functions,
-            sort_function,
+            func,
+        )
+
+        return new_partitions
+
+    @lazy_metadata_decorator(apply_axis="both")
+    def sort_by(
+        self,
+        axis: Union[int, Axis],
+        columns: Union[str, List[str]],
+        ascending: bool = True,
+        **kwargs,
+    ) -> "PandasDataframe":
+        """
+        Logically reorder rows (columns if axis=1) lexicographically by the data in a column or set of columns.
+
+        Parameters
+        ----------
+        axis : int or modin.core.dataframe.base.utils.Axis
+            The axis to perform the sort over.
+        columns : string or list
+            Column label(s) to use to determine lexicographical ordering.
+        ascending : boolean, default: True
+            Whether to sort in ascending or descending order.
+        **kwargs : dict
+            Keyword arguments to pass when sorting partitions.
+
+        Returns
+        -------
+        PandasDataframe
+            A new PandasDataframe sorted into lexicographical order by the specified column(s).
+        """
+        if not isinstance(columns, list):
+            columns = [columns]
+        # When we do a sort on the result of Series.value_counts, we don't rename the index until
+        # after everything is done, which causes an error when sorting the partitions, since the
+        # index and the column share the same name, when in actuality, the index's name should be
+        # None. This fixes the indexes name beforehand in that case, so that the sort works.
+
+        def sort_function(df):  # pragma: no cover
+            index_renaming = None
+            if any(name in df.columns for name in df.index.names):
+                index_renaming = df.index.names
+                df.index = df.index.set_names([None] * len(df.index.names))
+            df = df.sort_values(by=columns, ascending=ascending, **kwargs)
+            if index_renaming is not None:
+                df.index = df.index.set_names(index_renaming)
+            return df
+
+        axis = Axis(axis)
+        if axis != Axis.ROW_WISE:
+            raise NotImplementedError(
+                f"Algebra sort only implemented row-wise. {axis.name} sort not implemented yet!"
+            )
+
+        # If this df is empty, we don't want to try and shuffle or sort.
+        if len(self.axes[0]) == 0 or len(self.axes[1]) == 0:
+            return self.copy()
```

Comment:
I wonder if we can rework these conditions so that, when a sorting over unsupported column is requested, but the frame is actually empty, we just return a copy of self instead of failing... cc @RehanSD

Reply:
Well, I moved the condition a few lines above so it will be prior to the `if axis != ROW_AXIS` check.

However, I don't really understand what you meant under "sorting over an unsupported column". In my understanding we support all kinds of columns in sorting, don't we?

I may assume that you meant "grouping over an unsupported column", however, we can't just return a copy of self in case of grouping on an empty dataframe, there are still some actions to be done in such cases:
```python
>>> df = pd.DataFrame(index=[1, 1, 1, 0, 2, 0, 1])
Empty DataFrame
Columns: []
Index: [1, 1, 1, 0, 2, 0, 1]
>>> df.groupby(level=0).sum()
Empty DataFrame
Columns: []
Index: [0, 1, 2] # now it indicates groups
>>> df.groupby(level=0).size()
0    2
1    4
2    1
dtype: int64
```

Reply:
moreover, we claim that we don't support grouping on an empty frame for now and always default to pandas if it's the case